File: Java/Language/Basics/EnumTest.java
enum RiskLevel {
	NONE, LOW, HIGH;
}

class Investment {

	private double amount;
	private short years;
	private RiskLevel risk;

	public Investment(double p, short n) {
		amount = p;
		years = n;
		risk = RiskLevel.NONE;
	}

	public void allowRisk(boolean yes) { 
		risk = yes ? RiskLevel.LOW : RiskLevel.NONE;
	}

	public void adjustRisk(RiskLevel level) {
		risk = level;
	}

	public double income() {
		double rate;
		switch(risk){
			case LOW:
				rate = 7.5;
				break;
			case HIGH:
				rate = 8.5;
				break;
			default:
				rate = 6.0;
		}
		double cash =  amount * Math.pow(1 + rate / 100, years);
		return cash - amount;
	}
}


class EnumTest {

	public static void main(String[] args) {
		double i = Double.parseDouble(args[0]);
		short y = Short.parseShort(args[1]);
		Investment s = new Investment(i, y); 
		System.out.printf("Income in Silver scheme = %.2f%n", s.income());
		Investment g = new Investment(i, y);
		g.allowRisk(true); 
		System.out.printf("Income in Gold scheme = %.2f%n", g.income());
		Investment p = new Investment(i, y);
		p.adjustRisk(RiskLevel.HIGH);
		System.out.printf("Income in Platinum scheme = %.2f%n", p.income());
	}
}

File: CSharp/Basics/EnumTest.cs
using System;

enum RiskLevel {None, Low, High}

class Investment 
{
	private double invest;
	private int years;
	private RiskLevel risk;

	public Investment(double amount, int period)
	{
		invest = amount;
		years = period;
	}

	public void AllowRisk(bool yes)
	{
		risk = yes ? RiskLevel.Low : RiskLevel.None;
	}

	public void AdjustRisk(RiskLevel level)
	{
		risk = level;
	}

	public double GetIncome()
	{
		float rate;
		switch(risk)
		{
			case RiskLevel.Low:
				rate = 7;
				break;
			case RiskLevel.High:
				rate = 8;
				break;
			default:
				rate = 5;
				break;
		}
		double amount = invest * Math.Pow(1 + rate / 100, years);
		return amount - invest;
	}

}

static class Program 
{
	public static void Main(string[] args)
	{
		double p = double.Parse(args[0]);
		int n = int.Parse(args[1]);
		Investment inv = new Investment(p, n);
		Console.WriteLine("Income in Silver Scheme: {0:0.00}", inv.GetIncome());
		inv.AllowRisk(true);
		Console.WriteLine("Income in Gold Scheme: {0:0.00}", inv.GetIncome());
		inv.AdjustRisk(RiskLevel.High);
		Console.WriteLine("Income in Platinum Scheme: {0:0.00}", inv.GetIncome());
	}

}

